allennlp/modules/token_embedders/bag_of_word_counts_token_embedder.py

```diff
@@ -49,11 +64,13 @@ def forward(self,  # pylint: disable=arguments-differ
         ``(batch_size, vocab_size)``
         """
         bag_of_words_vectors = []
-
-        mask = get_text_field_mask({'tokens': inputs})
+        mask = get_text_field_mask({'tokens': inputs}, 1)
```

Comment:
wouldn't the logic be much cleaner if you did something here like

```
if self._ignore_oov:
    # also mask out positions corresponding to oov
    mask *= inputs != self._oov_idx
```
then you wouldn't have to do the `.nonzero().squeeze()` part below, which i find extremely non-obvious.